x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/rollup/PutRollupJobRequest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.rollup;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.fieldcaps.FieldCapabilities;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.protocol.xpack.rollup.job.RollupJobConfig;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class PutRollupJobRequest extends AcknowledgedRequest<PutRollupJobRequest> implements IndicesRequest, ToXContentObject {
+
+    private RollupJobConfig config;
+    private IndicesOptions indicesOptions = IndicesOptions.fromOptions(false, false, true, false);
+
+    public PutRollupJobRequest(final RollupJobConfig config) {
+        this.config = config;
+    }
+
+    public PutRollupJobRequest() {
```

Comment:
Do we need to worry about this empty ctor now?  E.g. if the user uses this instead of the other ctor we'll be throwing NPEs everywhere?